src/mmw/js/src/modeling/gwlfe/entry/calcs.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+var _ = require('lodash'),
+    turfArea = require('turf-area'),
+    App = require('../../../app');
+
+/**
+ * This module exports a set of objects for calculating outputs of, and auto-
+ * estimatates for, a manual entry field for GWLF-E.
+ *
+ * Each calculator object has two methods: `toOutput` which given a user
+ * value converts it to the expected GMS format, and `getAutoValue` that
+ * gets a user-friendly value from the GMS dataModel given a field name.
+ *
+ * Each manual entry field will be configured with one of these calculators.
+ */
+
+module.exports = {
+    // Takes the userValue and uses it directly in the output
+    Direct: {
+        toOutput: function(userValue) {
+            return userValue;
+        },
+        getAutoValue: function(fieldName, dataModel) {
+            return dataModel[fieldName];
+        },
+    },
+
+    // Takes the userValue and makes an array with the value repeated 12 times
+    Array12: {
+        toOutput: function(userValue) {
+            return [userValue, userValue, userValue,
+                    userValue, userValue, userValue,
+                    userValue, userValue, userValue,
+                    userValue, userValue, userValue];
+        },
+        getAutoValue: function(fieldName, dataModel) {
+            return dataModel[fieldName][0];
+        },
+    },
+
+    // Takes the userValue and divides into an array of 12 values evenly
+    EqualMonths: {
+        toOutput: function(userValue) {
+            var monthValue = userValue / 12;
+            return [monthValue, monthValue, monthValue,
+                    monthValue, monthValue, monthValue,
+                    monthValue, monthValue, monthValue,
+                    monthValue, monthValue, monthValue];
+        },
+        getAutoValue: function(fieldName, dataModel) {
+            return _.sum(dataModel[fieldName]);
+        },
+    },
+
+    // Takes the user value in MGD and converts it to cm/month, and applies it
+    // to twelve months. Equivalent to point_source_discharge method in
+    // mapshed/calcs.py
+    PointSourceDischarge: {
+        toOutput: function(userValue) {
+            var MONTHDAYS = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
+                AREA_SQM = turfArea(App.currentProject.get('area_of_interest')),
+                M3_PER_MGAL = 3785.41178,
+                CM_PER_M = 100.0;
+
+            return MONTHDAYS.map(function(DAYS_PER_MONTH) {
+                return userValue *
+                    (DAYS_PER_MONTH * M3_PER_MGAL * CM_PER_M / AREA_SQM);
```

Comment:
What does this represent? May be worth making it a named var before multiplying it with `userValue`.

Reply:
We're converting between MGD (million gallons per day) to cubic centimeters per month. I'll think of naming this cleaner, although the current naming is the same as:

https://github.com/WikiWatershed/model-my-watershed/blob/1e5376e5b06160b45cca3ac9aeaadc3bac68ee7e/src/mmw/apps/modeling/mapshed/calcs.py#L389-L390

Reply:
I adjusted the comments above this to be a bit more explicit, but decided to leave the code as is.

Reply:
Sounds good!